src/pseudo/PspTm.jl

```diff
@@ -0,0 +1,162 @@
+using SpecialFunctions: besselj
+using Interpolations: interpolate, Gridded, Linear
+using Plots: plot, plot!
+using Legendre: Plm
+
+struct PspTM <: NormConservingPsp
+    atomicNumber::Int                               # Atomic number of atom
+    valenceElectrons::Int                           # Number of valence electrons
+    lmax::Int                                       # Maximal angular momentum in the non-local part
+    lloc::Int                                       # Angular momentum used for local part
+    numGridPoints::Int                              # Number of grid points used in the radial grid
+    r2well::Float64                                 # Prefactor of a harmonic well
+    numProjectorFctns::Vector{Int}                  # Number of projector functions used for each angular momentum
+    pspCoreRadius::Vector{Float64}                  # Pseudopotential core radius per angular momentum
+    rms::Vector{Float64}                            # Measure of pseudopotential quality reflecting the value of the penalty function in designing the potential per angular momentum
+    energiesKB::Vector{Vector{Float64}}             # ekb are Kleinman-Bylander energies for each projection function for each angular momentum
+    epsatm::Vector{Float64}                         # The integral ∫^∞_0 4πr*(r*V(r)+valenceElectrons)dr
+    rchrg::Float64                                  # the Core charge radius for additional core charge used to match the xc contribution
+    fchrg::Float64                                  # The prefactor of the core charge expression
+    totCoreChrg::Float64                            # The total (integrated) core charge
+    pseudoPotentials::Vector{Vector{Float64}}       # Radial pseudopotential values for each angular momentum
+    firstProjectorVals::Vector{Vector{Float64}}     # Radial First projection functions for each angular momentum
+    secondProjectorVals::Vector{Vector{Float64}}    # If numProjectorFctns permits, the second projection functions for each angluar momentum
+    identifier::String                              # String identifying the PSP
+    description::String                             # Descriptive string
+end
+
+function parse_tm_file(path; identifier="")
+    file = readlines(path)
+    description = file[1]
+
```

Comment:
Just a general remark: In my experience debugging these parsers is a pain, especially if it's been a while since you wrote it. We had multiple subtle bugs in the HGH parser that only popped up a few months after I wrote it ... and until then you forget a lot. 

So to make it easier for our future selves: Now that you understand the format, I'd strongly suggest to comment in detail what stuff you are expecting to see in the TM file in each parsing step, what assumptions you are making about its structure and what you parse in each line.

I'd also avoid to try and be "clever" and use a lot of assertive programming. E.g. lines 32 and 34 are quite a mouthful and additionally might silently throw away fields beyond `6` without checking there are none. I'm not an expert in the TM format, so this might be completely ligit, but definitely comment on what you expect.